File: app/src/main/java/com/example/imac/httpurl/MainActivity.java
package com.example.imac.httpurl;

import android.content.Intent;
import android.net.Uri;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;


public class MainActivity extends AppCompatActivity {

    private UploadTask task;
    private TextView textView;
    // wordを入れる
    private EditText editText;

    // phpがPOSTで受け取ったwordを入れて作成するHTMLページ(適宜合わせてください)



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editText = findViewById(R.id.uriname);
        Button post = findViewById(R.id.post);

        // ボタンをタップして非同期処理を開始
        post.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String param0 = editText.getText().toString();

                if(param0.length() != 0){
                    textView = findViewById(R.id.get_num);
                    task = new UploadTask(textView);
                    task.execute(param0);
                }

            }
        });

        textView = findViewById(R.id.text_view);
    }

}

File: app/src/main/java/com/example/imac/httpurl/MainActivity.kt
package com.example.imac.httpurl

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

//a

class MainActivity : AppCompatActivity() {

    private var task: UploadTask? = null
    private var textView: TextView? = null
    // wordを入れる
    private var editText: EditText? = null

    // phpがPOSTで受け取ったwordを入れて作成するHTMLページ(適宜合わせてください)


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        editText = findViewById(R.id.uriname)
        val post = findViewById<Button>(R.id.post)

        // ボタンをタップして非同期処理を開始
        post.setOnClickListener {
            val param0 = editText!!.text.toString()

            if (param0.length != 0) {
                textView = findViewById(R.id.get_num)
                task = UploadTask(textView!!)
                task!!.execute(param0)
            }
        }

        textView = findViewById(R.id.text_view)
    }

}